packages/flutter/lib/src/widgets/router.dart

```diff
@@ -663,11 +691,13 @@ class _RouterState<T> extends State<Router<T>> {
 
   Future<void> _rebuild([void value]) {
     setState(() {/* routerDelegate is ready to rebuild */});
+    _routeInformation.value = _retrieveNewRouteInformation();
     return SynchronousFuture<void>(value);
   }
 
   void _handleRouterDelegateNotification() {
     setState(() {/* routerDelegate wants to rebuild */});
+    _routeInformation.value = _retrieveNewRouteInformation();
```

Comment:
This should probably should be in the method _maybeNeedToReportRouteInformation? That method will be called when the RouteInformation could have changed